nmrfx-processor/src/main/java/org/nmrfx/processor/datasets/Dataset.java

```diff
@@ -220,14 +220,13 @@ public Dataset(VecBase vector) {
         refUnits[0] = 3;
         rmsd = new double[1][1];
         values = new double[1][];
-        //System.err.println("Opened file " + fileName + " with " + this.nDim + " dimensions");
         addFile(fileName);
     }
 
     private Dataset(String fullName, String title,
             int[] dimSizes, boolean closeDataset) throws DatasetException {
-        //LOGGER.info("Make dataset {}", fullName);
         try {
+            // Cannot close this here as it is used to create other objects and is used in places outside this try
             RandomAccessFile raFile = new RandomAccessFile(fullName, "rw");
```

Comment:
I am not sure what to do with this resource. It cannot be closed here as it is used to create other objects, and is used later on outside of this try statement. When debugging with the tests it was closed later on here: https://github.com/nanalysis/nmrfx/blob/8b7ac8b8045c16e85aa7aad3a9bf0d1c17b783e0/nmrfx-processor/src/main/java/org/nmrfx/processor/datasets/Dataset.java#L474
Should I leave this sonar error unfixed? It seems like it would be a major refactor to change how this is setup.

Reply:
We could move it closer to usage at least, so that it doesn't get opened if layout is null. Other than that, I think we should keep it unfixed for now, or mark it as false positive if we're reasonably sure it gets closed on every usage.

Reply:
I moved it closer to its usage, I also added a try with a finally in copyDatasets to try and ensure that that raFile created in the private constructor is more likely to be closed. 
https://github.com/nanalysis/nmrfx/blob/dba89db353c37d4b13b35ddafc0d703c5d90f9a8/nmrfx-processor/src/main/java/org/nmrfx/processor/datasets/Dataset.java#L2062
There are also other constructors that set this variable that were not flagged by sonar as being unclosed, but I am not so confident that they get closed.